smart-contracts/contracts/mixins/MixinTransfer.sol

```diff
@@ -132,6 +132,8 @@ contract MixinTransfer is
     if (toKey.tokenId == 0) {
       toKey.tokenId = _tokenId;
       _recordOwner(_recipient, _tokenId);
+      // Clear any previous approvals
+      _clearApproval(_tokenId);
```

Comment:
Movin this call to _`clearApproval` to here to avoid calling twice for expired keys, while still ensuring approvals are cleared when necessary.